### PR TITLE
Warn when `getStaticPaths` exists without a `prerender` statement

### LINKED ISSUE #6350: Prerendering from Content Collections still adds content to SSR server code
### What version of `astro` are you using?

2.0.14

### Are you using an SSR adapter? If so, which one?

Deno

### What package manager are you using?

npm

### What operating system are you using?

Linux, Ubuntu 22.04

### Describe the Bug

When prerendering content from content collections using the method described in the documentation and used in the blog example, the content of each item/page is still added to the entry.mjs file. Standard pages are prerendered to html and then not added to entry.mjs.

Duplicating the already prerendered content inside entry.mjs leads to very large server files and is in general not efficient. 

See the example provided and you can see how the output from blog posts in the content folder are in entry.mjs but the content of the about.astro page is no longer included there when using prerendering.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-8mbuj5

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

- Resolves #6350
- Moves the warning for `getStaticPaths` existing without `prerender`.
- Previously, this warning happened at runtime during `dev`. Now the warning happens at `compile` time during `dev` and `build`.
- Works for both `server | hybrid` outputs.

## Testing

We don't have (reliable) tests for CLI logs, unfortunately

Tested manually

## Docs

cc @withastro/maintainers-docs for feedback on the warning message. These warnings aren't something we document like our errors, but they are user-facing!

**Before**
```sh
[getStaticPaths] getStaticPaths() in /src/pages/posts/[slug].astro is ignored when "output: server" is set.
```

**After**
```sh
[getStaticPaths] The getStaticPaths() statement in /src/pages/posts/[slug].astro has been ignored.

Add `export const prerender = true;` to prerender this page.
```

### CODE REVIEW

**Review by natemoo-re — COMMENTED** [review]
Just tweaking the `getStaticPaths` without `prerender` warning so it runs during `dev` and `build`. Previously, it only ran when a page was being rendered in `dev`.

File: packages/astro/src/vite-plugin-scanner/index.ts [hunk 1]
```diff
@@ -26,11 +35,16 @@ export default function astroScannerPlugin({ settings }: { settings: AstroSettin
...
+			const pageOptions = await scan(code, id, settings);
 
 			if (typeof pageOptions.prerender === 'undefined') {
 				pageOptions.prerender = defaultPrerender;
 			}
+			
+			// `getStaticPaths` warning is just a string check, should be good enough for most cases
+			if (!pageOptions.prerender && isServerLikeOutput(settings.config) && code.includes('getStaticPaths')) {
+				warn(logging, "getStaticPaths", `The getStaticPaths() statement in ${bold(rootRelativePath(settings.config.root, fileURL, true))} has been ignored.\n\nAdd \`export const prerender = true;\` to prerender this page.`);
+			}
```
[the remark this answers is not in the document]
Comment (natemoo-re) [reply packages/astro/src/vite-plugin-scanner/index.ts]: Updated this one per @sarah11918's feedback that mentioning `output: "${mode}"` would be helpful.

**Review by bluwy — APPROVED** [review]
I like the move to compile/build-time warnings!

**Review by delucis — APPROVED** [review]
I like the new warning! Nice that it includes a helpful actionable step of how to fix.

**Review by matthewp — APPROVED** [review]

**Pull request merged**